chat-plugins/calculator.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+function isNumeric(str) {
+	return !isNaN(parseFloat(str)) && isFinite(str);
+}
+
+function clean(arr) {
+	let i;
+	for (i = 0; i < arr.length; i++) {
+		if (arr[i] === "") {
+			arr.splice(i, 1);
+		}
+	}
+	return arr;
+}
+
+function parseMathematicalExpression(infix) {
+	// Shunnting-yard Algorithm -- https://en.wikipedia.org/wiki/Shunting-yard_algorithm
+	let outputQueue = "";
```

Comment:
You can just make this an array of tokens; you're passing it to `solveRPN` anyway.

Reply:
Was this a performance decision? Resizing arrays are in general better-performing than stringbuffers in JavaScript.